File: module-main/src/main/java/com/yyxnb/module_main/bean/MainHomeBean.java
package com.yyxnb.module_main.bean;

import com.yyxnb.what.core.interfaces.IData;

public class MainHomeBean implements IData<Long> {
    public int id;
    public int type;
    public String title;
    public String des;
    public String url;
    public String color;
    public int span;

    @Override
    public int id() {
        return id;
    }

    @Override
    public String getCode() {
        return null;
    }

    @Override
    public String getMsg() {
        return null;
    }

    @Override
    public Long getResult() {
        return null;
    }

    @Override
    public boolean isSuccess() {
        return false;
    }
}

File: module_main/src/main/java/com/yyxnb/module_main/bean/MainHomeBean.kt
package com.yyxnb.module_main.bean

import com.yyxnb.widget.interfaces.IData

data class MainHomeBean(
        @JvmField
        var id: Int = 0,
        var type: Int = 0,
        var title: String? = "",
        var des: String? = "",
        var url: String? = "",
        var color: String? = "",
        @JvmField
        var span: Int = 0
) : IData<Long> {

    override fun id(): Int {
        return id
    }

    override fun getCode(): String {
        return id.toString()
    }

    override fun getMsg(): String? {
        return null
    }

    override fun getResult(): Long? {
        return null
    }

    override fun isSuccess(): Boolean {
        return false
    }
}